### PR TITLE
Linter fixes for plugins/inputs/[de]*

### PR DESCRIPTION
**Pull request by zak-pawel**
Following findings were fixed:
```
plugins/inputs/dcos/client.go:236:57                             revive       import-shadowing: The name 'url' shadows an import name
plugins/inputs/dcos/client.go:262:23                             revive       import-shadowing: The name 'url' shadows an import name
plugins/inputs/dcos/client.go:276:52                             revive       import-shadowing: The name 'url' shadows an import name
plugins/inputs/dcos/client.go:321:25                             revive       import-shadowing: The name 'url' shadows an import name
plugins/inputs/dcos/client.go:322:2                              revive       import-shadowing: The name 'url' shadows an import name
plugins/inputs/dcos/dcos.go:355:2                                revive       import-shadowing: The name 'url' shadows an import name
plugins/inputs/directory_monitor/directory_monitor.go:278:43     revive       flag-parameter: parameter 'firstLine' seems to be a control flag, avoid control coupling
plugins/inputs/directory_monitor/directory_monitor_test.go:79:5  ineffassign  ineffectual assignment to err
plugins/inputs/disk/disk.go:18:2                                 revive       confusing-naming: Field 'MountPoints' differs only by capitalization to other field in the struct type DiskStats
plugins/inputs/disk/disk_test.go:113:2                           ineffassign  ineffectual assignment to err
plugins/inputs/disk/disk_test.go:118:2                           ineffassign  ineffectual assignment to err
plugins/inputs/disk/disk_test.go:1:9                             revive       import-shadowing: The name 'disk' shadows an import name
plugins/inputs/disk/disk_test.go:368:2                           ineffassign  ineffectual assignment to err
plugins/inputs/disk/disk_test.go:373:2                           ineffassign  ineffectual assignment to err
plugins/inputs/diskio/diskio.go:77:4                             revive       import-shadowing: The name 'filter' shadows an import name
plugins/inputs/diskio/diskio_linux_test.go:110:8                 revive       defer: prefer not to defer chains of function calls
plugins/inputs/diskio/diskio_linux_test.go:98:3                  revive       defer: prefer not to defer inside loops
plugins/inputs/disque/disque.go:68:3                             revive       import-shadowing: The name 'url' shadows an import name
plugins/inputs/dns_query/dns_query.go:19:2                       staticcheck  SA9004: only the first constant in this group has an explicit type
plugins/inputs/docker/client.go:1:9                              revive       import-shadowing: The name 'docker' shadows an import name
plugins/inputs/docker/docker.go:1:9                              revive       import-shadowing: The name 'docker' shadows an import name
plugins/inputs/docker/docker.go:636:1                            revive       argument-limit: maximum number of arguments per function exceeded; max 6 but got 7
plugins/inputs/docker/docker.go:880:26                           revive       flag-parameter: parameter 'perDevice' seems to be a control flag, avoid control coupling
plugins/inputs/docker/docker.go:968:2                            revive       import-shadowing: The name 'filter' shadows an import name
plugins/inputs/docker/docker.go:977:2                            revive       import-shadowing: The name 'filter' shadows an import name
plugins/inputs/docker/docker.go:989:2                            revive       import-shadowing: The name 'filter' shadows an import name
plugins/inputs/docker_log/docker_log.go:310:20                   revive       unnecessary-stmt: switch with only one case can be replaced by an if-then
plugins/inputs/docker_log/docker_log.go:424:2                    revive       import-shadowing: The name 'filter' shadows an import name
plugins/inputs/docker_log/docker_log.go:433:2                    revive       import-shadowing: The name 'filter' shadows an import name
plugins/inputs/docker_log/docker_log.go:445:2                    revive       import-shadowing: The name 'filter' shadows an import name
plugins/inputs/ecs/ecs.go:223:2                                  revive       import-shadowing: The name 'filter' shadows an import name
plugins/inputs/ecs/ecs.go:232:2                                  revive       import-shadowing: The name 'filter' shadows an import name
plugins/inputs/ecs/ecs.go:253:2                                  revive       import-shadowing: The name 'filter' shadows an import name
plugins/inputs/elasticsearch/elasticsearch.go:647:8              revive       bool-literal-in-expr: omit Boolean literal in expression
plugins/inputs/elasticsearch/elasticsearch_test.go:209:2         revive       unexported-naming: the symbol IsMasterResultTokens is local, its name should start with a lowercase letter
plugins/inputs/elasticsearch/elasticsearch_test.go:247:2         revive       unexported-naming: the symbol IsNotMasterResultTokens is local, its name should start with a lowercase letter
plugins/inputs/ethtool/ethtool_linux.go:3:9                      revive       import-shadowing: The name 'ethtool' shadows an import name
plugins/inputs/ethtool/ethtool_test.go:38:3                      revive       bare-return: avoid using bare returns, please add return expressions
plugins/inputs/ethtool/ethtool_test.go:73:3                      revive       bare-return: avoid using bare returns, please add return expressions
plugins/inputs/eventhub_consumer/eventhub_consumer.go:1:9        revive       import-shadowing: The name 'eventhub' shadows an import name
plugins/inputs/exec/exec.go:1:9                                  revive       import-shadowing: The name 'exec' shadows an import name
plugins/inputs/execd/execd_test.go:142:39                        revive       import-shadowing: The name 'metric' shadows an import name

```

### CODE REVIEW

File: plugins/inputs/diskio/diskio_linux_test.go [hunk 1]
```diff
@@ -92,12 +92,14 @@ func TestDiskIOStats_diskName(t *testing.T) {
...
-		name, _ := s.diskName("null")
-		require.Equal(t, tc.expected, name, "Templates: %#v", tc.templates)
+		func() {
+			s := DiskIO{
+				NameTemplates: tc.templates,
+			}
+			defer setupNullDisk(t, &s, "null")() //nolint:revive // done on purpose, cleaning will be executed properly
+			name, _ := s.diskName("null")
+			require.Equal(t, tc.expected, name, "Templates: %#v", tc.templates)
+		}()
```
Comment (srebhan) [inline plugins/inputs/diskio/diskio_linux_test.go:102]: Would
```golang
	for _, tc := range tests {
		s := DiskIO{
			NameTemplates: tc.templates,
		}
		name, _ := s.diskName("null")	
		require.Equal(t, tc.expected, name, "Templates: %#v", tc.templates)
		setupNullDisk(t, &amp;s, "null")()
	}
```
also be an option?

File: plugins/inputs/directory_monitor/directory_monitor.go [hunk 2]
```diff
@@ -256,16 +256,14 @@ func (monitor *DirectoryMonitor) ingestFile(filePath string) error {
 
 func (monitor *DirectoryMonitor) parseFile(parser parsers.Parser, reader io.Reader) error {
 	// Read the file line-by-line and parse with the configured parse method.
-	firstLine := true
+	lineNumber := 0
 	scanner := bufio.NewScanner(reader)
 	for scanner.Scan() {
-		metrics, err := monitor.parseLine(parser, scanner.Bytes(), firstLine)
+		metrics, err := monitor.parseLine(parser, scanner.Bytes(), lineNumber)
```
Comment (srebhan) [inline plugins/inputs/directory_monitor/directory_monitor.go:262]: This is a pure workaround around the linter. You are doing the same as before just silencing the linter by exploiting its dumbness. :-) Shall we really do this or better switch off that rule?

**Review by srebhan — COMMENTED** [review]
Looks good to me.

I have only one general comment about `flag-parameter: parameter 'perDevice' seems to be a control flag, avoid control coupling`. I'm still not sure what a good replacement pattern is for those cases. Your "fix" is just a smart exploitation of the linter weaknesses, but you are still doing the same as before... :-)

File: plugins/inputs/directory_monitor/directory_monitor.go (hunk 2 above)
Comment (zak-pawel) [reply plugins/inputs/directory_monitor/directory_monitor.go:262]: I know :) I don't have better idea, how to fix this in the better way without reorganizing a lot of code. 
Rule itself is not that dumb, for other cases (in this PR and previous ones) it was pretty easy to fix this kind of finding (see: https://github.com/influxdata/telegraf/pull/9379/files#diff-8f6f90da1087a523fbc9bb69df6736bf68545e6d9fef07b5537b9ce22ff4a257L886 )

Comment (zak-pawel) [reply plugins/inputs/directory_monitor/directory_monitor.go]: @srebhan 
`flag-parameter` rule disabled, `directory_monitor.go` reverted

**Review by reimda — APPROVED** [review]

**Review by srebhan — APPROVED** [review]
LGTM.

**Pull request merged**